app/Http/Controllers/CampaignsController.php

```diff
@@ -43,22 +43,29 @@ public function __construct(Registrar $registrar, CampaignService $campaignServi
     public function index()
     {
         $ids = $this->campaignService->getCampaignIdsFromSignups();
+
         $campaigns = $this->campaignService->findAll($ids);
-        $campaigns = $this->campaignService->groupByCause($campaigns);
+
+        $campaigns = $campaigns->map(function($campaign, $key) {
+            if ($campaign) {
+                $campaign = $this->campaignService->getCampaignPostStatusCounts($campaign);
```

Comment:
This seems worrisome since it'll make N database queries (one per campaign ID)... can we rephrase this query as a query to the signups table that's then grouped or filtered by the given campaign IDs?